--- a/linux/vdma/memory.c
+++ b/linux/vdma/memory.c
@@ -167,9 +167,11 @@
     }

     if (HAILO_DMA_DMABUF_BUFFER != buffer_type) {
+        mmap_read_lock(current->mm);
         vma = find_vma(current->mm, addr_or_fd);
         if (IS_ENABLED(HAILO_SUPPORT_MMIO_DMA_MAPPING)) {
             if (NULL == vma) {
+                mmap_read_unlock(current->mm);
                 dev_err(dev, "no vma for virt_addr/size = 0x%08lx/0x%08zx\n", addr_or_fd, size);
                 ret = -EFAULT;
                 goto cleanup;
@@ -181,6 +183,7 @@
             buffer_type = HAILO_DMA_DMABUF_BUFFER;
             ret = create_fd_from_vma(dev, vma);
             if (ret < 0) {
+                mmap_read_unlock(current->mm);
                 dev_err(dev, "Failed creating fd from vma in given dmabuf\n");
                 goto cleanup;
             }
@@ -188,6 +191,7 @@
             dmabuf_from_pointer_addr = addr_or_fd;
             addr_or_fd = ret;
         }
+        mmap_read_unlock(current->mm);
     }

     // TODO: is MMIO DMA MAPPINGS STILL needed after dmabuf
